src/TSTransformer/nodes/transformMethodDeclaration.ts

```diff
@@ -33,7 +33,11 @@ export function transformMethodDeclaration(
 	luau.list.pushList(statements, transformStatementList(state, node.body.statements));
 
 	let name = transformPropertyName(state, node.name);
-	if (ts.hasDecorators(node) && !luau.isSimple(name)) {
+
+	if (
+		(ts.hasDecorators(node) || node.parameters.some(parameter => ts.hasDecorators(parameter))) &&
+		!luau.isSimple(name)
+	) {
```

Comment:
This check effectively duplicates conditions from elsewhere to keep behaviour the same (setting `classElementObjectKey` as opposed to re-generating the key). This is fragile if the conditions ever change. It would be better to find a solution which uses the existing prereq logic to determine whether it's necessary to prereq the key statements.

Reply:
Do you have any suggestions as to how this could be solved? I'm not very familiar with prereqs and don't have a test case for this behaviour.

Reply:
It's a tricky one to solve. I'll have a go at my local project, see if I come up with any easy strategies.

Reply:
Actually, why do we do this partial setting at all? What if we just do a `pushToVarIfComplex` and put all the keys in the map?